webapp/src/app/mytokens/page.tsx

```diff
@@ -107,16 +109,22 @@ function BatchRedeemCard({batch}:{batch: Batch}) {
     enabled: sameAddresses !== undefined,
     args: [sameAddresses!, batch.token_ids as unknown as bigint[]]
   });
-  console.log("🚀 ~ file: page.tsx:110 ~ BatchRedeemCard ~ ownedIndex:", ownedIndex)
 
-  const filteredArray = useCallback(() => {
+  const ownerTokens = useCallback(() => {
     if(ownedIndex) {
       return batch.token_ids.filter((_, index) => +ownedIndex[index].toString() === 1);
     }
   }, [batch.token_ids, ownedIndex]);
-  console.log("🚀 ~ file: page.tsx:117 ~ filteredArray ~ filteredArray:", filteredArray())
+ 
+  const { config } = usePrepareContractWrite({
```

Comment:
note: somehow `redeemTokens` still gets triggered on load, with no args, and reverts.

Reply:
updated conditions